components/channel_notifications_modal/channel_notifications_modal.tsx

```diff
@@ -35,279 +64,301 @@ type Props = PropsFromRedux & {
-        let ignoreChannelMentionsDefault: ChannelNotifyProps['ignore_channel_mentions'] = IgnoreChannelMentions.OFF;
-
-        if (channelMemberNotifyProps?.mark_unread === NotificationLevels.MENTION || (currentUserNotifyProps.channel && currentUserNotifyProps.channel === 'false')) {
-            ignoreChannelMentionsDefault = IgnoreChannelMentions.ON;
-        }
-
-        let ignoreChannelMentions = channelMemberNotifyProps?.ignore_channel_mentions;
-        if (!ignoreChannelMentions || ignoreChannelMentions === IgnoreChannelMentions.DEFAULT) {
-            ignoreChannelMentions = ignoreChannelMentionsDefault;
-        }
-
-        return {
-            desktopNotifyLevel: channelMemberNotifyProps?.desktop || NotificationLevels.DEFAULT,
-            desktopThreadsNotifyLevel: channelMemberNotifyProps?.desktop_threads || NotificationLevels.ALL,
-            markUnreadNotifyLevel: channelMemberNotifyProps?.mark_unread || NotificationLevels.ALL,
-            pushNotifyLevel: channelMemberNotifyProps?.push || NotificationLevels.DEFAULT,
-            pushThreadsNotifyLevel: channelMemberNotifyProps?.push_threads || NotificationLevels.ALL,
-            ignoreChannelMentions,
-        };
-    }
+    const desktop = channelMemberNotifyProps?.desktop === NotificationLevels.DEFAULT ? currentUserNotifyProps.desktop : (channelMemberNotifyProps?.desktop || currentUserNotifyProps.desktop);
+    const push = channelMemberNotifyProps?.push === NotificationLevels.DEFAULT ? currentUserNotifyProps.desktop : (channelMemberNotifyProps?.push || currentUserNotifyProps.push);
+
+    return {
+        desktop,
+        desktop_threads: channelMemberNotifyProps?.desktop_threads || NotificationLevels.ALL,
+        mark_unread: channelMemberNotifyProps?.mark_unread || NotificationLevels.ALL,
+        push,
+        push_threads: channelMemberNotifyProps?.push_threads || NotificationLevels.ALL,
+        ignore_channel_mentions: ignoreChannelMentions,
+        channel_auto_follow_threads: channelMemberNotifyProps?.channel_auto_follow_threads || 'false',
+    };
+}
 
-    handleHide = () => this.setState({show: false});
+type SettingsType = {
+    desktop: ChannelNotifyProps['desktop'];
+    channel_auto_follow_threads: 'true' | 'false';
+    desktop_threads: ChannelNotifyProps['desktop_threads'];
+    mark_unread: ChannelNotifyProps['mark_unread'];
+    push: ChannelNotifyProps['push'];
+    push_threads: ChannelNotifyProps['push_threads'];
+    ignore_channel_mentions: ChannelNotifyProps['ignore_channel_mentions'];
+};
 
-    handleExit = () => {
-        this.updateSection(NotificationSections.NONE);
-        this.props.onExited();
-    }
+export default function ChannelNotificationsModal(props: Props) {
+    const {formatMessage} = useIntl();
+    const [show, setShow] = useState(true);
+    const [serverError, setServerError] = useState('');
+    const [channelNotifyProps] = useState(props.channelMember && props.channelMember.notify_props);
```

Comment:
Do we need to assign these props to the state? I worry that mixing them might cause side effects and mutations we don't expect. How about we just pass the props wherever this is needed